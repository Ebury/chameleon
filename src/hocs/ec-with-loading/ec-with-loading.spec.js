import { mount, createLocalVue, createWrapper } from '@vue/test-utils';
import withLoading from './ec-with-loading';

describe('EcWithLoading', () => {
  function mountEcWithLoading(props, mountOpts) {
    const localVue = createLocalVue();

    const Component = localVue.extend({
      props: {
        customProp: {
          type: Number,
          default: 1,
        },
      },
      render(h) {
        return h('div', { attrs: { 'data-custom': this.customProp } });
      },
    });

    const hocWrapper = mount(withLoading(Component), {
      localVue,
      propsData: { ...props },
      ...mountOpts,
    });

    const componentWrapper = createWrapper(hocWrapper.vm.$children[0].$vnode);

    return { hocWrapper, componentWrapper };
  }

  it('should render properly', () => {
    const { hocWrapper } = mountEcWithLoading();
    expect(hocWrapper.element).toMatchSnapshot();
  });

  describe(':props', () => {
    it('should hide loading when isLoading prop is not set', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: undefined });
      expect(hocWrapper.element).toMatchSnapshot();
    });

    it('should show loading when isLoading prop is set to true', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: true });
      expect(hocWrapper.element).toMatchSnapshot();
    });

    it('should hide loading when isLoading prop is set to false', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: false });
      expect(hocWrapper.element).toMatchSnapshot();
    });

    it('should render transparent by default', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: true, isLoadingTransparent: undefined });
      expect(hocWrapper.element).toMatchSnapshot();
    });

    it('should render transparent when isLoadingTransparent is set to true', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: true, isLoadingTransparent: true });
      expect(hocWrapper.element).toMatchSnapshot();
    });

    it('should not render transparent when isLoadingTransparent is set to false', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: true, isLoadingTransparent: false });
      expect(hocWrapper.element).toMatchSnapshot();
    });

    it('should use iconSize prop', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: true, loadingIconSize: 64 });
      expect(hocWrapper.element).toMatchSnapshot();
    });

    it('should pass rest of the props to the wrapped component', () => {
      const { hocWrapper } = mountEcWithLoading({ isLoading: true, customProp: 2 });
      expect(hocWrapper.element).toMatchSnapshot();
    });
  });
});
