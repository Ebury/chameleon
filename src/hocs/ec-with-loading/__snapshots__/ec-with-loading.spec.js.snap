// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcWithLoading :props should hide loading when isLoading prop is set to false 1`] = `
<div
  class="ec-loading"
  data-test="ec-loading"
>
  <!---->
   
  <div
    class="ec-loading__content ec-loading__content--is-transparent"
    data-test="ec-loading__content"
  >
    <div
      data-custom="1"
    />
  </div>
</div>
`;

exports[`EcWithLoading :props should not render transparent when isLoadingTransparent is set to false 1`] = `
<div
  class="ec-loading"
  data-test="ec-loading"
>
  <div
    class="ec-loading__backdrop"
    data-test="ec-loading__backdrop"
  >
    <svg
      class="ec-loading__icon ec-icon"
      data-test="ec-loading__icon"
      height="48"
      width="48"
    >
      <use
        xlink:href="#ec-simple-loading"
      />
    </svg>
  </div>
   
  <div
    class="ec-loading__content"
    data-test="ec-loading__content"
  >
    <div
      data-custom="1"
    />
  </div>
</div>
`;

exports[`EcWithLoading :props should pass rest of the props to the wrapped component 1`] = `
<div
  class="ec-loading"
  data-test="ec-loading"
>
  <div
    class="ec-loading__backdrop ec-loading__backdrop--is-transparent"
    data-test="ec-loading__backdrop"
  >
    <svg
      class="ec-loading__icon ec-icon"
      data-test="ec-loading__icon"
      height="48"
      width="48"
    >
      <use
        xlink:href="#ec-simple-loading"
      />
    </svg>
  </div>
   
  <div
    class="ec-loading__content ec-loading__content--is-transparent"
    data-test="ec-loading__content"
  >
    <div
      data-custom="2"
    />
  </div>
</div>
`;

exports[`EcWithLoading :props should render transparent by default 1`] = `
<div
  class="ec-loading"
  data-test="ec-loading"
>
  <div
    class="ec-loading__backdrop ec-loading__backdrop--is-transparent"
    data-test="ec-loading__backdrop"
  >
    <svg
      class="ec-loading__icon ec-icon"
      data-test="ec-loading__icon"
      height="48"
      width="48"
    >
      <use
        xlink:href="#ec-simple-loading"
      />
    </svg>
  </div>
   
  <div
    class="ec-loading__content ec-loading__content--is-transparent"
    data-test="ec-loading__content"
  >
    <div
      data-custom="1"
    />
  </div>
</div>
`;

exports[`EcWithLoading :props should render transparent when isLoadingTransparent is set to true 1`] = `
<div
  class="ec-loading"
  data-test="ec-loading"
>
  <div
    class="ec-loading__backdrop ec-loading__backdrop--is-transparent"
    data-test="ec-loading__backdrop"
  >
    <svg
      class="ec-loading__icon ec-icon"
      data-test="ec-loading__icon"
      height="48"
      width="48"
    >
      <use
        xlink:href="#ec-simple-loading"
      />
    </svg>
  </div>
   
  <div
    class="ec-loading__content ec-loading__content--is-transparent"
    data-test="ec-loading__content"
  >
    <div
      data-custom="1"
    />
  </div>
</div>
`;

exports[`EcWithLoading :props should show loading when isLoading prop is set to true 1`] = `
<div
  class="ec-loading"
  data-test="ec-loading"
>
  <div
    class="ec-loading__backdrop ec-loading__backdrop--is-transparent"
    data-test="ec-loading__backdrop"
  >
    <svg
      class="ec-loading__icon ec-icon"
      data-test="ec-loading__icon"
      height="48"
      width="48"
    >
      <use
        xlink:href="#ec-simple-loading"
      />
    </svg>
  </div>
   
  <div
    class="ec-loading__content ec-loading__content--is-transparent"
    data-test="ec-loading__content"
  >
    <div
      data-custom="1"
    />
  </div>
</div>
`;

exports[`EcWithLoading :props should throw an error when isLoading prop is not set 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[Vue warn]: Invalid prop: type check failed for prop \\"show\\". Expected Boolean, got Undefined 

found in

---> <EcLoading>
       <Root>",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`EcWithLoading :props should use iconSize prop 1`] = `
<div
  class="ec-loading"
  data-test="ec-loading"
>
  <div
    class="ec-loading__backdrop ec-loading__backdrop--is-transparent"
    data-test="ec-loading__backdrop"
  >
    <svg
      class="ec-loading__icon ec-icon"
      data-test="ec-loading__icon"
      height="64"
      width="64"
    >
      <use
        xlink:href="#ec-simple-loading"
      />
    </svg>
  </div>
   
  <div
    class="ec-loading__content ec-loading__content--is-transparent"
    data-test="ec-loading__content"
  >
    <div
      data-custom="1"
    />
  </div>
</div>
`;
